src/istore_agg.c

```diff
@@ -3,18 +3,33 @@
-                                                                                             \
-    if (PG_ARGISNULL(1) && PG_ARGISNULL(0)) PG_RETURN_NULL();                                \
-                                                                                             \
-    state = PG_ARGISNULL(0) ? state_init(agg_context) : (ISAggState *) PG_GETARG_POINTER(0); \
-                                                                                             \
-    if (PG_ARGISNULL(1)) PG_RETURN_POINTER(state);                                           \
+#define BIG_ISTORE 1
+#define ISTORE 0
+
+#define RETURN_AGG_TRANSFN(_istore, _agg)                                                        \
+    do {                                                                                         \
+        MemoryContext  agg_context;                                                              \
+        ISAggState    *state;                                                                    \
+                                                                                                 \
+        if (!AggCheckCallContext(fcinfo, &agg_context))                                          \
+            elog(ERROR, "aggregate function called in non-aggregate context");                   \
+                                                                                                 \
+        if (PG_ARGISNULL(1) && PG_ARGISNULL(0)) PG_RETURN_NULL();                                \
+                                                                                                 \
+        state = PG_ARGISNULL(0) ? state_init(agg_context) : (ISAggState *) PG_GETARG_POINTER(0); \
+                                                                                                 \
+        if (PG_ARGISNULL(1)) PG_RETURN_POINTER(state);                                           \
+                                                                                                 \
+        switch (_istore)                                                                         \
+        {                                                                                        \
+            case BIG_ISTORE:                                                                     \
+                PG_RETURN_POINTER(bigistore_agg_internal(state, PG_GETARG_BIGIS(1), _agg));      \
+            case ISTORE:                                                                         \
+                PG_RETURN_POINTER(istore_agg_internal(state, PG_GETARG_IS(1), _agg));            \
+            default:                                                                             \
+                elog(ERROR, "unexpected istore type passed.");                                   \
+        }                                                                                        \
+    } while(0)
```

Comment:
so `RETURN_AGG_TRANSFN(2, _agg)`
would compile to to a function without any return statement which the compiler would error right ?
which would be the same as having no default at all